torchtext/datasets/enwik9.py

```diff
@@ -26,9 +26,9 @@
 @_create_dataset_directory(dataset_name=DATASET_NAME)
 @_wrap_split_argument(('train',))
 def EnWik9(root, split):
+    logging.info('Creating {} data'.format(split))
     dataset_tar = download_from_url(URL, root=root, hash_value=MD5, hash_type='md5')
     extracted_files = extract_archive(dataset_tar)
     path = extracted_files[0]
-    logging.info('Creating {} data'.format(split))
     return _RawTextIterableDataset(DATASET_NAME,
-                                   NUM_LINES[split], _read_text_iterator(path))
```

Comment:
Since this is BC-breaking and removes this data irrevocably and by default, should we offer the transform separately as a functional and to be optionally applied by the user via e.g.
```
data = EnWik9(split='train')
data = _create_data_from_wikipedia_xml_dumps(data)
```
(we might want to pick a different name for the functional then)

Reply:
Yes, I like the idea of functional. This way we can also avoid putting additional options if the user wanted the raw data without filtering.